language/vm/vm-genesis/src/lib.rs

```diff
@@ -391,6 +422,24 @@ pub fn encode_genesis_transaction_with_validator(
                 "Validator set in emitted event does not match validator set fed into genesis transaction"
             );
 
+            // (5) The fourth event should be the discovery set change event
```

Comment:
are (3)/(4) checks that will be added by the `TODO` below?

Reply:
the TODO below will test the same thing as (4), just with reading the emitted discovery set and checking that it matches what we fed in.